Tensile/Source/lib/include/Tensile/MatchingLibrary.hpp

```diff
@@ -71,9 +78,22 @@ namespace Tensile
 
             auto closestEntry = table->findBestMatch(problem, transform);
```

Comment:
Thread-safety?

Here's rocBLAS's implementation of [double-checked locking](https://en.wikipedia.org/wiki/Double-checked_locking):

https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/98ac69fcc8f406ff35b96d63af9072862c564e17/library/src/include/logging.h#L276